app/components/App/App.js

```diff
@@ -0,0 +1,207 @@
+        isAllowedToBeSteppedOn: true,
+        numberOfStepInPath: -1,
+      };
+    }
+  }
+
+  return board;
+}
+
+function cloneBoard(board) {
+  return (board.map(row => (row.map(cell => ({ ...cell, })))));
+}
+
+@autobind
+export default class App extends React.Component {
+  static get defaultProps() {
+    return {
+      dimensions: {
+        x: 12,
+        y: 12,
+      },
+      colors: {
+        A: 'red',
+        B: 'blue',
+        C: 'green',
+      },
+    };
+  }
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      colorA: props.colors.A,
+      colorB: props.colors.B,
+      colorC: props.colors.C,
+      board: resetBoard(props.dimensions.x, props.dimensions.y),
+      pathStartingPoint: [],
+      pathEndingPoint: [],
+      startButtonPressed: false,
+      endButtonPressed: false,
+      // pathExistsFlag: false, // 'true if a path was found'
+      // time: '',
+    };
+  }
+
+  setBoardColor(updatedColor, newColor) {
+    this.setState(() => {
+      const color = {};
+      color[updatedColor] = newColor;
+      return color;
+    });
+  }
+
+  setStartLocation(x, y) {
+    this.setState({
+      pathStartingPoint: [ x, y ],
+      startButtonPressed: false,
+    });
+  }
+
+  setEndLocation(x, y) {
+    this.setState(() => ({
+      pathEndingPoint: [ x, y ],
+      endButtonPressed: false,
+    }));
+  }
+
+  handleClickOnBoard(x, y) {
+    if (this.state.startButtonPressed) {
+      return this.setStartLocation(x, y);
+    }
+
+    if (this.state.endButtonPressed) {
+      return this.setEndLocation(x, y);
+    }
+
+    return this.toggleCellAllowedToBeSteppedOn(x, y);
+  }
+
+  toggleCellAllowedToBeSteppedOn(x, y) {
+    console.log('toggling cell state');
+    const { board, } = this.state;
+    const boardCopy = cloneBoard(board);
+    boardCopy[y][x].isAllowedToBeSteppedOn = !this.state.board[y][x].isAllowedToBeSteppedOn;
+    console.log('board copy:');
+    console.log(boardCopy);
+    this.setState({ board: boardCopy, });
+  }
+
+  toggleStartButtonPressed() {
+    this.setState({
+      startButtonPressed: !this.state.startButtonPressed,
+      endButtonPressed: false,
+    });
+  }
+
+  toggleEndButtonPressed() {
+    this.setState({
+      endButtonPressed: !this.state.endButtonPressed,
+      startButtonPressed: false,
+    });
+  }
+
+  findAndDrawPath() {
+    const { pathStartingPoint, pathEndingPoint, board, } = this.state;
+    const boardCopy = cloneBoard(board);
+    const findPathParameters = {
+      field: boardCopy,
+      startPoint: pathStartingPoint,
+      endPoint: pathEndingPoint,
+    };
+    const path = findPath(findPathParameters);
+    const stepCounter = -1;
+    this.drawPath(path, stepCounter);
+  }
+
+  drawPath(pathRemaining, stepNumber) {
+    if (pathRemaining.length === 0) {
+      return;
+    }
+    this.addStepToPathOnBoard(pathRemaining[0], stepNumber + 1);
+    window.setTimeout(this.drawPath.bind(this, pathRemaining.slice(1), stepNumber + 1), 500);
```

Comment:
So why do you need the timeout?

Reply:
right, because I want to draw the path incrementally. Gives a feel of "walking the path"

Reply:
So the purpose is animation?

Reply:
indeed